### PR TITLE
feat: worker delete

### PR DESCRIPTION
**Pull request by benPearce1**
[sc-29479]

depends on https://github.com/OctopusDeploy/go-octopusdeploy/pull/163

image

### CODE REVIEW

File: pkg/cmd/worker/delete/delete.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+			if err != nil {
+				return err
+			}
+
+			if worker == nil {
+				return fmt.Errorf("cannot find a worker with name or ID of '%s'", idOrName)
+			}
+
+			if !skipConfirmation { // TODO NO_PROMPT env var or whatever we do there
+				return question.DeleteWithConfirmation(f.Ask, "deployment target", worker.Name, worker.GetID(), func() error {
```
Comment (jbristowe) [inline pkg/cmd/worker/delete/delete.go:61]:
```suggestion
				return question.DeleteWithConfirmation(f.Ask, "worker", worker.Name, worker.GetID(), func() error {
```

File: pkg/cmd/worker/delete/delete.go [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+					return delete(opts.Client, worker)
+				})
+			}
+
+			return delete(opts.Client, worker)
+		},
+	}
+
+	question.RegisterConfirmDeletionFlag(cmd, &skipConfirmation, "deployment target")
```
Comment (jbristowe) [inline pkg/cmd/worker/delete/delete.go:70]:
```suggestion
	question.RegisterConfirmDeletionFlag(cmd, &skipConfirmation, "worker")
```

File: pkg/cmd/worker/delete/delete.go (hunk 1 above)
Comment (benPearce1) [reply pkg/cmd/worker/delete/delete.go:61]: ffs

**Review by jbristowe — CHANGES_REQUESTED** [review]
Minor edits, otherwise LGTM.

**Review by jbristowe — APPROVED** [review]

**Pull request merged**